recipes/_agent_common_service.rb

```diff
@@ -2,6 +2,15 @@
 # For windows, installing and starting service by commands, https://www.zabbix.com/documentation/3.4/manual/appendix/install/windows_agent
 # Also need to configure firewall to let Zabbix server talk to agent.
 if platform_family?('windows')
+  powershell_script 'stop_zabbix_if_exist' do
+    code <<-EOH
+    net stop 'Zabbix Agent' 2>$1 | out-null;
+    sc.exe delete "Zabbix Agent" 2>$1 | out-null;
+    Get-Process zabbix_agentd | Stop-Process -Force 2>$1 | out-null;
+    exit 0;
+    EOH
+    action :nothing
+  end
```

Comment:
We have a `notifies :restart, 'service[zabbix_agentd]'` for Linux that handles this without crazy Powershell hackery.  Look at [this open-source Zabbix Windows cookbook](https://github.com/thomasvincent/zabbix_windows_chef/blob/master/recipes/windows_agent.rb). I see no reason our cookbook can't be as simple as that for Windows.

There is a [service resource](https://docs.chef.io/resource_service.html) in Chef. Any reason we don't use it? Why do we do everything manually?

Reply:
Because we're using a much older cookbook https://github.com/laradji/zabbix for both linux and windows,  chef fail to install and start the Zabbix service with `service` resource on Windows. That's why I'm doing it manually.